clients/java/src/main/java/io/camunda/client/CamundaClientBuilder.java

```diff
@@ -0,0 +1,220 @@
+   * store.
+   */
+  CamundaClientBuilder caCertificatePath(String certificatePath);
+
+  /**
+   * A custom {@link CredentialsProvider} which will be used to apply authentication credentials to
+   * requests.
+   */
+  CamundaClientBuilder credentialsProvider(CredentialsProvider credentialsProvider);
+
+  /** Time interval between keep alive messages sent to the gateway. The default is 45 seconds. */
+  CamundaClientBuilder keepAlive(Duration keepAlive);
+
+  CamundaClientBuilder withInterceptors(ClientInterceptor... interceptor);
+
+  CamundaClientBuilder withJsonMapper(JsonMapper jsonMapper);
+
+  /**
+   * Overrides the authority used with TLS virtual hosting. Specifically, to override hostname
+   * verification in the TLS handshake. It does not change what host is actually connected to.
+   *
+   * <p>This method is intended for testing, but may safely be used outside of tests as an
+   * alternative to DNS overrides.
+   *
+   * <p>This setting does nothing if a {@link #usePlaintext() plaintext} connection is used.
+   *
+   * @param authority The alternative authority to use, commonly in the form <code>host</code> or
+   *     <code>host:port</code>
+   * @apiNote For the full definition of authority see [RFC 2396: Uniform Resource Identifiers
+   *     (URI): Generic Syntax](http://www.ietf.org/rfc/rfc2396.txt)
+   */
+  CamundaClientBuilder overrideAuthority(String authority);
+
+  /**
+   * A custom maxMessageSize allows the client to receive larger or smaller responses from Zeebe.
+   * Technically, it specifies the maxInboundMessageSize of the gRPC channel. The default is 4194304
+   * = 4MB.
+   */
+  CamundaClientBuilder maxMessageSize(int maxSize);
+
+  /**
+   * A custom streamEnabled allows the client to use job stream instead of job poll. The default
+   * value is set as enabled.
+   */
+  CamundaClientBuilder defaultJobWorkerStreamEnabled(boolean streamEnabled);
+
+  /**
+   * If enabled, the client will make use of the default retry policy defined. False by default.
+   *
+   * <p>NOTE: the default retry policy is taken from the {@code gateway-service-config.json} in the
+   * {@code io.camunda:zeebe-gateway-protocol-impl} JAR.
+   */
+  CamundaClientBuilder useDefaultRetryPolicy(final boolean useDefaultRetryPolicy);
+
+  /**
+   * If true, will prefer to use REST over gRPC for calls which can be done over both REST and gRPC.
+   * This is an experimental API which is present while we migrate the bulk of the API from gRPC to
+   * REST. Once done, this will also be removed.
+   *
+   * <p>NOTE: not all calls can be done over REST (or HTTP/1) yet, this is also subject to change.
+   *
+   * @param preferRestOverGrpc if true, the client will use REST instead of gRPC whenever possible
+   * @deprecated since 8.5, will be removed in 8.8
+   * @return this builder for chaining
+   */
+  @ExperimentalApi("https://github.com/camunda/zeebe/issues/16166")
```

Comment:
🔧 The deprecation warning from the `Zeebe...` class is still needed here. It might make sense to double-check the other methods in all classes that had something like this.

Reply:
I need to double check all the deprecation yes 👍